storage/src/test/java/org/apache/kafka/storage/internals/log/LogSegmentTest.java

```diff
@@ -202,13 +202,8 @@ public void testReadWhenNoMaxPosition(boolean minOneMessage) throws IOException
             MemoryRecords ms = records(50, "hello", "there");
             seg.append(51, RecordBatch.NO_TIMESTAMP, -1L, ms);
 
-            // read before first offset
```

Comment:
Not sure if this case is relevant. There's only 1 batch which starts at offset 50, so the returned `LogOffsetMetadata` has an offset field of 50 rather than 48. Under `LogSegment#read`'s previous behavior, offset 48 is just directly returned, so I don't think this case is actually doing anything.